### ISSUE TITLE
Not able to search any song or artist on Mac

### ISSUE THREAD

**Issue by aladine** [mihaiolteanu/versuri#3]
Hi Mihai,
This is a great package. However, I could not run versuri-display it on my emacs:
OS version: Mac OS 10.13.6
Emacs version: Version 27.0.50

The error is 
> esqlite--error: esqlite: "Currently this system cannot handle async family."

**Comment by mihaiolteanu** [comment]
https://github.com/mhayashi1120/Emacs-esqlite/blob/08a779a821f8d32c1a1985d8d9eb6cf21646ce2e/esqlite.el#L357

That's an error from the esqlite package itself.
You get it when `versuri` tries to save your searched lyrics in the db.
It seems like it doesn't and it won't ever work on Mac.

If you have any ideas how to fix this, I'm all ears.

**Comment by mihaiolteanu** [comment]
Hello Dan,
Did you manage anything with this?

**Comment by aladine** [comment]
I have no solution until now. 
The issue for esqlite was already mentioned in https://github.com/mhayashi1120/Emacs-esqlite/issues/2
Async process for macos is buggy when using with esqlite.

**Comment by isamert** [comment]
I have to use Mac for my daily job, so I looked into this issue. I replaced stream functions with their file counterparts based on the comment that the esqlite author made in the linked issue, it seems to work:

```
esqlite-stream-execute -> esqlite-execute
esqlite-stream-read -> esqlite-read
```

I can put up an PR that makes versuri to use different family of functions based on current operating system or I can simply replace all functions just as I did right now. I didn't notice any slowdown although my use case with versuri is quite limited. What would be your take on this, would you consider accepting an PR like this @mihaiolteanu?

**Comment by mihaiolteanu** [comment]
Ideally all would work fine regardless of the OS. I would be happy with that.

> I can simply replace all functions just as I did right now

You mean also replacing the implementation around them, right?

I haven't used the package myself in a while so it would take some time to review any PR, but any fixes to confirmed bugs are more than welcome, of course!

Thanks for looking into it!

**Comment by isamert** [comment]
> You mean also replacing the implementation around them, right?

Yep, I will prepare a PR soon and we can discuss there.

> I haven't used the package myself in a while so it would take some time to review any PR

No worries, I just patched it locally and using it on Mac, so no rush needed for me.

**Comment by mihaiolteanu** [comment]
@aladine Can you please check the [PR](https://github.com/mihaiolteanu/versuri/pull/11) from @isamert and let me know if it fixes your issue?

@isamert I'm sorry this takes so long. I would idealy need some feedback from other users, as well.

**Comment by isamert** [comment]
> I would idealy need some feedback from other users, as well.

Yep, I see. That's the right choice, the fix might not even work for others because I prepared the fix based on a comment that the esqilite package author made and it happened to work for me. Better get some feedback from others to ensure that it's working for everyone.
